src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -501,8 +499,8 @@ FetchTableCommon(text *tableSchemaNameText, text *tableNameText, uint64 remoteTa
 							   " do not match", nodeNameCount, nodePortCount)));
 	}
 
-	schemaName = text_to_cstring(tableSchemaNameText);
-	tableName = text_to_cstring(tableNameText);
+	relationNameList = textToQualifiedNameList(tableNameText);
+	DeconstructQualifiedName(relationNameList, &schemaName, &tableName);
```

Comment:
You need schema name here because you introduced schema parameters in PR #653. If you follow function calls, you will see that you always concatenate schema name with relation name. So, here it would be enough to use  ResolveRelationId() on the table name to get relation id. Also reverting back the other function calls will make the code simpler.

There is just makeRangeVar() inside FetchRegularTable() where you need schema name separately, you can postpone to call DeconstructQualifiedName() until there.